driver/src/main/java/org/neo4j/driver/internal/bolt/basicimpl/async/connection/ChannelConnectedListener.java

```diff
@@ -66,11 +66,11 @@ public void operationComplete(ChannelFuture future) {
                         error = new ServiceUnavailableException(
                                 String.format("Unable to write Bolt handshake to %s.", this.address), error);
                     }
-                    this.handshakeCompletedPromise.setFailure(error);
+                    this.handshakeCompletedFuture.completeExceptionally(error);
                 }
             });
         } else {
-            handshakeCompletedPromise.setFailure(databaseUnavailableError(address, future.cause()));
+            handshakeCompletedFuture.completeExceptionally(databaseUnavailableError(address, future.cause()));
```

Comment:
Is it possible to craft a unit test that prevents a regression? I.e., fails with the old code, passes with the new code by simulating a failing channel. Or is there too much coupling between netty and the driver here?